### PR TITLE
Fixes some missing includes/declaring budgets and if->ifdef

### PR DESCRIPTION
**Pull request by amznestebanpapp**
Signed-off-by: Esteban Papp <81431996+amznestebanpapp@users.noreply.github.com>

### CODE REVIEW

File: Gems/Atom/RHI/Code/Include/Atom/RHI/MemorySubAllocator.h [hunk 1]
```diff
@@ -7,11 +7,14 @@
...
 #pragma once
 
+#include <AzCore/Debug/Profiler.h>
 #include <Atom/RHI/FreeListAllocator.h>
 #include <Atom/RHI/PoolAllocator.h>
 #include <Atom/RHI/MemoryAllocation.h>
 #include <Atom/RHI.Reflect/MemoryEnums.h>
 
+AZ_DECLARE_BUDGET(RHI);
+
```
Comment (rgba16f) [inline Gems/Atom/RHI/Code/Include/Atom/RHI/MemorySubAllocator.h:17]: Is this needed? My understanding was that the budget just had to be declared once over the whole codebase.

**Review by rgba16f — APPROVED** [review]

File: Gems/Atom/RHI/Code/Include/Atom/RHI/MemorySubAllocator.h (hunk 1 above)
Comment (amznestebanpapp) [reply Gems/Atom/RHI/Code/Include/Atom/RHI/MemorySubAllocator.h:17]: it has to be defined once, but you still need to have that symbol visible.
Under the hood the macro uses a function to get that budget

`AZ_DECLARE_BUDGET` -> declares the function
`AZ_DEFINE_BUDGET` -> defines the function
`AZ_PROFILE_FUNCTION` -> calls the function, so it needs to be able to see the declaration

**Review by AMZN-daimini — APPROVED** [review]

**Comment by AMZN-daimini** [comment]
Note that @sphrose already submitted a fix for the include issue in debug overnight: https://github.com/o3de/o3de/pull/6246

**Comment by amznestebanpapp** [comment]
> Note that @sphrose already submitted a fix for the include issue in debug overnight: #6246

@sphrose fix was just on the define, not the includes. Trivial merge to solve so I will just merge this.

**Comment by AMZN-daimini** [comment]
> > Note that @sphrose already submitted a fix for the include issue in debug overnight: #6246
> 
> @sphrose fix was just on the define, not the includes. Trivial merge to solve so I will just merge this.

Yes, I was thinking define and wrote include for some reason, sorry :D

**Pull request merged**